iroh/examples/transfer.rs

```diff
@@ -404,3 +472,16 @@ fn parse_byte_size(s: &str) -> Result<u64> {
     let cfg = parse_size::Config::new().with_binary();
     cfg.parse_size(s).map_err(|e| anyhow::anyhow!(e))
 }
+
+fn watch_conn_type(endpoint: &Endpoint, node_id: NodeId) -> AbortOnDropHandle<()> {
+    let mut stream = endpoint.conn_type(node_id).unwrap().stream();
```

Comment:
Not a PR review comment, rather out API here isn't very nice.  It seems this `Result` that is unwrapped only exists because the `NodeId` might not be known.

1. At the very least this should be an `Option`, not a result.
2. Why not make it return pending forever?  Or rather until that node becomes known.  This could be either a footgun or a nice feature that you can start watching before the NodeId is known to the magic socket.  And because a NodeId can be "forgotten" by the magic socket, and that doesn't suddenly result in an error at this point, it already behaves like this.

Reply:
Turning this into an issue: https://github.com/n0-computer/iroh/issues/3300